### PR TITLE
[SR-2694] Cannot JSON serialize array with boolean value 

### PR DESCRIPTION
**Pull request by saiHemak**
https://bugs.swift.org/browse/SR-2694

### CODE REVIEW

**Comment by modocache** [comment]
LGTM, but I defer to people who know corelibs-foundation well, like @phausler or @parkera.

@swift-ci please test

**Comment by modocache** [comment]
@swift-ci please test

**Comment by modocache** [comment]
@swift-ci Please test

**Comment by modocache** [comment]
@swift-ci Please test

File: Foundation/NSJSONSerialization.swift [hunk 1]
```diff
@@ -285,6 +285,8 @@ private struct JSONWriter {
             try serializeDictionary(dict)
         } else if let null = obj as? NSNull {
             try serializeNull(null)
+        } else if let boolVal = obj as? Bool {
+            try serializeNumber(NSNumber(booleanLiteral: boolVal))
```
Comment (phausler) [inline Foundation/NSJSONSerialization.swift]: NSNumber(value: boolVal) would be better since you are not passing a literal here

**Comment by saiHemak** [comment]
@phausler ..Thanks for the review..I have modified the  code as per your comments .Please review

File: TestFoundation/TestNSJSONSerialization.swift [hunk 2]
```diff
@@ -928,6 +929,16 @@ extension TestNSJSONSerialization {
         XCTAssertThrowsError(try JSONSerialization.writeJSONObject(str, toStream: outputStream, options: []))
     }
     
+    func test_booleanJSONObject() {
+        do {
+            let mydata = try JSONSerialization.data(withJSONObject: [true])
+            XCTAssertEqual(String(data: mydata, encoding: String.Encoding.utf8),"[1]")
```
Comment (pushkarnk) [inline TestFoundation/TestNSJSONSerialization.swift]: nit: missing whitespace between arguments

**Comment by pushkarnk** [comment]
@swift-ci please test

**Comment by pushkarnk** [comment]
@swift-ci please test

**Comment by parkera** [comment]
@swift-ci please test

**Comment by parkera** [comment]
@phausler if you're ok with this after the changes, please merge

**Pull request merged**